sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/models/SendOptions.java

```diff
@@ -54,6 +56,33 @@ public String getPartitionKey() {
         return partitionKey;
     }
 
+    /**
+     * Gets the identifier of the Event Hub partition that the {@link EventData events} will be sent to. If the
+     * identifier is not specified, the Event Hubs service will be responsible for routing events that are sent to an
+     * available partition.
+     *
+     * @return The identifier of the Event Hub partition that the {@link EventData events} will be set to. {@code null}
+     *     or an empty string if Event Hubs service is responsible for routing events.
+     */
+    public String getPartitionId() {
```

Comment:
Both `SendOptions` and `BatchOptions` must have a check to either use partitionKey or partition Id and not both.

Reply:
The validation happens when you go to send the message through AsyncProducer.send(). I wanted to delay validation until they go to invoke those.